onmt/inputters/dynamic_iterator.py

```diff
@@ -309,6 +330,15 @@ def build_dynamic_dataset_iter(opt, transforms_cls, vocabs, copy=False,
                                  pin_memory=True,
                                  multiprocessing_context="fork",
                                  num_workers=data_iter.num_workers,
-                                 prefetch_factor=50,
-                                 worker_init_fn=data_iter._init_datasets)
+                                 worker_init_fn=data_iter._init_datasets,
+                                 prefetch_factor=opt.prefetch_factor)
+        # We disable automatic batching in the DataLoader.
```

Comment:
If we want to document to this extent then I would prefer to pout this in the docstring of the method so that all the content will be reflected in the actual documentation.
thanks